nifi-nar-bundles/nifi-hadoop-libraries-bundle/nifi-hadoop-libraries-nar/pom.xml

```diff
@@ -21,19 +21,70 @@
         <maven.javadoc.skip>true</maven.javadoc.skip>
         <source.skip>true</source.skip>
     </properties>
-    <dependencies>
-        <dependency>
+    <profiles>
+      <profile>
+        <id>vanilla</id>
+   <activation>
+      <activeByDefault>true</activeByDefault>
+    </activation>
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.nifi</groupId>
+                <artifactId>nifi-standard-services-api-nar</artifactId>
+                <type>nar</type>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.hadoop</groupId>
+                <artifactId>hadoop-client</artifactId>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.avro</groupId>
+                <artifactId>avro</artifactId>
+            </dependency>
+        </dependencies>
+      </profile>
+      <profile>
+        <id>custom_hadoop_libraries</id>
```

Comment:
found the culprit. hadoop.version is used by the hadoop poms and that causes the conflict (spark adds the repo's to the top pom.xml, hence no conflict).

Renaming it to `nifi.hadoop.version` fixes it.